### PR TITLE
add test case and bug fix

### PR DESCRIPTION
**Pull request by yangyuliufeng**
Signed-off-by: yangyuliufeng <qlw705706@gmail.com>

### Ⅰ. Describe what this PR does
add test case of tieredStorage
fix bugs when out of bound

### Ⅴ. Special notes for reviews

### CODE REVIEW

File: pkg/utils/tieredstore/tiered_store.go [hunk 1]
```diff
@@ -20,26 +20,36 @@ import (
...
+func (s sortMediumType) Len() int {
+	return len(s)
 }
 
-type sortMediumType []common.MediumType
+func (s sortMediumType) Swap(i, j int) {
+	if i < s.Len() && j < s.Len() {
+		s[i], s[j] = s[j], s[i]
+	} else {
+		log.Info("index out of bound, will not swap", "i", i, "j", j)
```
Comment (allenhaozi) [inline pkg/utils/tieredstore/tiered_store.go:39]: suggest:
If the index overflows, it explicitly returns an error,
This way the caller knows there is a problem with the index

Comment (yangyuliufeng) [reply pkg/utils/tieredstore/tiered_store.go:39]: > suggest:
> If the index overflows, it explicitly returns an error,
> This way the caller knows there is a problem with the index

This function is an implement of an interface, so I cannot change the return.

Comment (cheyang) [reply pkg/utils/tieredstore/tiered_store.go:39]: I think we don't need to check index overflow. Please see https://yourbasic.org/golang/how-to-sort-in-go/ . And its implementation in https://github.com/golang/go/blob/master/src/sort/sort.go#L231

**Review by cheyang — CHANGES_REQUESTED** [review]

File: pkg/utils/tieredstore/tiered_store.go (hunk 1 above)
Comment (xieydd) [reply pkg/utils/tieredstore/tiered_store.go:39]: @yangyuliufeng Have some known error?

**Review by cheyang — APPROVED** [review]

**Pull request merged**